database/skylink.go

```diff
@@ -0,0 +1,94 @@
+package database
+
+import (
+	"context"
+	"regexp"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var (
+	skylinkRE *regexp.Regexp = nil
+)
+
+// Skylink represents a skylink object in the DB.
+type Skylink struct {
+	ID      primitive.ObjectID `bson:"_id,omitempty" json:"-"`
+	Skylink string             `bson:"skylink" json:"skylink"`
+	Size    int64              `bson:"size" json:"size"`
+}
+
+// Skylink gets the DB object for the given skylink.
+// If it doesn't exist it creates it.
+func (db *DB) Skylink(ctx context.Context, skylink string) (*Skylink, error) {
+	skylinkHash, err := validateSkylink(skylink)
+	if err != nil {
+		return nil, ErrInvalidSkylink
+	}
+	// Provisional skylink object.
+	skylinkRec := Skylink{
+		Skylink: skylinkHash,
+	}
+	// Try to find the skylink in the database.
+	filter := bson.D{{"skylink", skylinkHash}}
+	sr := db.staticSkylinks.FindOne(ctx, filter)
+	err = sr.Decode(&skylinkRec)
+	if err == mongo.ErrNoDocuments {
+		// It's not there, insert it.
+		var ior *mongo.InsertOneResult
+		ior, err = db.staticSkylinks.InsertOne(ctx, skylinkRec)
```

Comment:
I think this'll be subject to duplciate key exceptions if two connections are inserting the same skylink at the same time. You could use `upsert` to circumvent this issue, and I think we should.

Reply:
Good point. Added.